lightcycle-api/src/main/java/com/soundcloud/lightcycle/FragmentLightCycle.java

```diff
@@ -1,29 +1,24 @@
 package com.soundcloud.lightcycle;
 
-import android.annotation.TargetApi;
 import android.app.Activity;
-import android.app.Fragment;
 import android.content.Context;
-import android.os.Build;
 import android.os.Bundle;
 import android.view.MenuItem;
 import android.view.View;
 
-@TargetApi(Build.VERSION_CODES.HONEYCOMB)
-public interface FragmentLightCycle<T extends Fragment> {
-    void onAttach(T fragment, Activity activity);
-    @TargetApi(23)
-    void onAttach(T fragment, Context context);
-    void onCreate(T fragment, Bundle bundle);
-    void onViewCreated(T fragment, View view, Bundle savedInstanceState);
-    void onActivityCreated(T fragment, Bundle bundle);
-    void onStart(T fragment);
-    void onResume(T fragment);
-    boolean onOptionsItemSelected(T fragment, MenuItem item);
-    void onPause(T fragment);
-    void onStop(T fragment);
-    void onSaveInstanceState(T fragment, Bundle bundle);
-    void onDestroyView(T fragment);
-    void onDestroy(T fragment);
-    void onDetach(T fragment);
+public interface FragmentLightCycle<T> {
```

Comment:
I don't think it's a must, but I think it's better to have it.  
Correct me if I'm wrong but even the LightCycles doesn't need to know the real type of the HostType, but the HostType still need to be an instance of Fragment right?

For example if I change the `LicensePresenter` in `real-world` example to implement  `FragmentLightCycle<LicenseView>` instead of `SupportFragmentLightCycle<LicenseView>` it will fails.
  
I think because it will still fail in the compile time after the annotation processor created the generated method it won't be a big problem, but I think it'll be better if the user can know their mistake faster if we put `@TargetApi` annotation

Another reason is to make it uniform with another implementation [here](https://github.com/soundcloud/lightcycle/blob/cc8630bab20209cc82d1b5a2da1d7ff2e7cc11e7/lightcycle-api/src/main/java/com/soundcloud/lightcycle/DefaultFragmentLightCycle.java#L11), [here](https://github.com/soundcloud/lightcycle/blob/cc8630bab20209cc82d1b5a2da1d7ff2e7cc11e7/lightcycle-api/src/main/java/com/soundcloud/lightcycle/LightCycles.java#L111) and [here](https://github.com/soundcloud/lightcycle/blob/cc8630bab20209cc82d1b5a2da1d7ff2e7cc11e7/lightcycle-lib/src/main/java/com/soundcloud/lightcycle/FragmentLightCycleDispatcher.java#L17)